Sources/KlaviyoSwift/StateManagement/StateManagement.swift

```diff
@@ -352,7 +358,25 @@ struct KlaviyoReducer: ReducerProtocol {
             return .run { [numAttempts] send in
                 let result = await environment.klaviyoAPI.send(request, numAttempts)
                 switch result {
-                case .success:
+                case let .success(data):
+                    do {
+                        switch request.endpoint {
+                        case .fetchForms:
+                            let formsResponse = try JSONDecoder().decode(FullFormsResponse.self, from: data)
+                            await send(.handleFormsResponse(formsResponse))
+                        default:
+                            break
+                        }
+                    } catch is DecodingError {
+                        let error = KlaviyoAPIError.dataDecodingError(request)
+                        await send(handleRequestError(request: request, error: error, retryInfo: nil))
```

Comment:
Small thing: Let's not use this function here and instead just return dequeue and possibly log it out.